Okta.AspNet.Abstractions/UserAgentHandler.cs

```diff
@@ -4,20 +4,35 @@
 // </copyright>
 
 using System;
+using System.Net;
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Okta.AspNet.Abstractions
 {
+    // TODO: consider renaming this class to OktaHttpMessageHandler, to clarify to future contributors the (expanding) responsibility of this class
     public class UserAgentHandler : DelegatingHandler
     {
-        private Lazy<string> _userAgent;
+        private readonly Lazy<string> _userAgent;
 
-        public UserAgentHandler(string frameworkName, Version frameworkversion)
+        public UserAgentHandler(string frameworkName, Version frameworkVersion, OktaWebOptions oktaWebOptions = null)
         {
-            InnerHandler = new HttpClientHandler();
-            _userAgent = new Lazy<string>(() => new UserAgentBuilder(frameworkName, frameworkversion).GetUserAgent());
+            _userAgent = new Lazy<string>(() => new UserAgentBuilder(frameworkName, frameworkVersion).GetUserAgent());
+            if (string.IsNullOrEmpty(oktaWebOptions?.ProxyAddress))
```

Comment:
Question: [I think there's a scenario where you might want to use the system proxy (in which case ProxyAddress can be null or empty) plus credentials](https://mcguirev10.com/2019/12/19/httpclient-web-proxy-authentication.html#proxy-aware-handler). Should we consider that too?

Reply:
That feels out of scope for this change, I could be wrong.  Possibly/probably something that should be considered, but I tried to ensure that the change here only changed behavior if a `ProxyAddress` was specified, otherwise existing behavior "should" be the default.